bin/install

```diff
@@ -34,12 +34,20 @@ install_java() {
 	local versionstr=$1
 	local destdir=$2
 
-	IFS=- read distro version <<< $versionstr
+	IFS=- read -r distro version <<< "${versionstr}"
 
 	local variant=$(get_variant $distro)
 	[[ -z "$ASDF_JAVA_ERROR" ]] || return
 
-	get_java $distro $version $variant $destdir
+	# Perform all operations is a temporary directory that will be
+	# automatically cleaned up on exit (success or failure).
+	local wrkdir
```

Comment:
I would do `local wrkdir=$(mktemp -dt asdf-java.XXXXXX)`...all in one line :sunglasses:

Reply:
Although it is a bit theoretical at this point, combining declaration and assignment masks return value of the `mktemp` command. Shellcheck warns about this, see [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155). As I said I know the return value isn't used right now, but I'm hoping `set -e` could be enabled eventually which would make it useful.

Reply:
You are so right! Didn't actually know about that...

Reply:
I'll say, just when I thought I knew most quirks in bash, thanks to shellcheck I discovered a whole bunch more :wink: tis a never ending source of astonishment :joy: